From 92b30aa23fbd30ccf717aba31fdb016f76397849 Mon Sep 17 00:00:00 2001
From: Alex Deymo <deymo@chromium.org>
Date: Wed, 20 Mar 2013 15:12:28 -0700
Subject: [PATCH] BlueZ 4.99: Fix CreatePairedDevice and DiscoverServices
 concurrence bug

The second step during pairing to a device with CreatePairedDevice before
send the message return to DBus caller involves a service discover step.
This step uses the same infrastructure (browse_req) than DiscoverServices
without checking if it is currently used.

This fix both cases, when DiscoverServices is called during a
CreatePairedDevice.

diff --git a/src/device.c b/src/device.c
index 648f93f..c863f8b 100644
--- a/src/device.c
+++ b/src/device.c
@@ -738,6 +738,9 @@ static DBusMessage *discover_services(DBusConnection *conn,
 	if (device->browse)
 		return btd_error_in_progress(msg);

+	if (device->bonding)
+		return btd_error_in_progress(msg);
+
 	if (dbus_message_get_args(msg, NULL, DBUS_TYPE_STRING, &pattern,
 						DBUS_TYPE_INVALID) == FALSE)
 		return btd_error_invalid_args(msg);
